modules/grpc/bigquery/bigquery-worker.h

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2023 László Várady
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 as published
+ * by the Free Software Foundation, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#ifndef BIGQUERY_WORKER_H
+#define BIGQUERY_WORKER_H
+
+#include "compat/cpp-start.h"
+
+#include "syslog-ng.h"
+#include "logthrdest/logthrdestdrv.h"
+
+typedef struct _BigQueryDestWorker BigQueryDestWorker;
+
+LogThreadedDestWorker *bigquery_dw_new(LogThreadedDestDriver *o, gint worker_index);
```

Comment:
[just a comment]

For future reference, we do not need to have a `...-worker.h` (we don't need to make the C wrapper and C ctor public), as it is only used in the driver's C ctor, which can instead delegate this to the driver's C++ code. I only found this out recently, when I implemented the `syslog-ng-otlp()` destination, where the worker's construct had to be overridden by the C++ class.

See:
* https://github.com/syslog-ng/syslog-ng/blob/master/modules/grpc/otel/otel-dest.cpp#L171
* https://github.com/syslog-ng/syslog-ng/blob/master/modules/grpc/otel/otel-dest.cpp#L75
* https://github.com/syslog-ng/syslog-ng/blob/master/modules/grpc/otel/syslog-ng-otlp-dest.cpp#L54

The implementation here is 100% correct, and I do not ask you to change it, I just thought I mention this little detail regarding our C++ support.